### PR TITLE
fix: [M3-7191] - Missing primary and secondary action button labels in Delete SSH Key dialog and Clone Domain drawer

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝
This PR aimed to fix a bug with a missing button label in the Delete SSH Key dialog and ended up fixing another related bug in the Clone Domain drawer.

## Major Changes 🔄
- Adds the missing `Delete` label to the primary action button in the SSH key delete dialog
- Makes `label` a required, not an optional, prop for `ActionButtonProps`
- Inadvertently fixes another bug, where the `Cancel` button appears invisible in the Clone Domain drawer because it was also missing a label
- Adds unit test for `DeleteSSHKeyDialog` component and updates unit test for `ActionPanel` component.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| image | image |
| <video src="https://github.com/linode/manager/assets/114685994/edafc430-bc59-4bff-871c-e03e9992c2fd"/> | <video src="https://github.com/linode/manager/assets/114685994/2458a1b4-da0b-4609-addb-5e61c537dff6" /> |

## How to test 🧪
1. **How to setup test environment?**
- If you do not already have an SSH key on your account, you will need to create one via Profile > `Add an SSH Key`. 
- If you do not already have a domain on your account, you will need to create one via Domains > `Create Domain`. 
- Checkout this PR.
- `yarn dev`
2. **How to reproduce the issue (if applicable)?**
- Go to https://cloud.linode.com/profile/keys.
- Click on the Delete action for your SSH key.
- Observe that the Delete SSH Key dialog will pop up, but it is missing a button label for the primary action button `Delete`.
- Go to https://cloud.linode.com/domains.
- Click on the Clone action for your existing domain.
- Observe in the Clone drawer that the user can hover over the area of the secondary action `Cancel` button and click it to close the drawer, but it is otherwise invisible.
3. **How to verify changes?**
- Go to http://localhost:3000/profile/keys.
- Click on the Delete action for your SSH key.
- Observe that the Delete SSH Key dialog will pop up, and the `Delete` button is labeled accordingly.
- Go to http://localhost:3000/domains.
- Click on the Clone action for your existing domain. 
- Observe in the Clone drawer that a labeled `Cancel` button is visible and clicking it closes the drawer.
4. **How to run Unit or E2E tests?**
```
yarn test DeleteSSHKeyDialog.test.tsx ActionPanel.test.tsx
```

### CODE REVIEW

File: packages/manager/src/components/ActionsPanel/ActionsPanel.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { Box, BoxProps } from '../Box';
 interface ActionButtonsProps extends ButtonProps {
   'data-node-idx'?: number;
   'data-testid'?: string;
-  label?: string;
+  label: string;
```
Comment (mjac0bs) [inline packages/manager/src/components/ActionsPanel/ActionsPanel.tsx:13]: This prop was optional, which is what allowed the bug to exist in the first place. We shouldn't allow unlabeled buttons.

When I made `label` required, it led to an error at the one other place in the app where a action button was missing a label: the Clone Drawer and its secondary button, `Cancel`. This was another bug, with an existing but unlabeled and therefore invisible-seeming button.

**Comment by tyler-akamai** [comment]
Tested the following:
- [ ] The Delete SSH Key dialog pops up, and the Delete button is labeled accordingly
- [ ] The Domain Clone drawer has a Cancel button and clicking it closes the drawer
- [ ] DeleteSSHKeyDialog.test.tsx and ActionPanel.test.tsx passes

**Review by tyler-akamai — APPROVED** [review]

File: packages/manager/src/components/ActionsPanel/ActionsPanel.tsx (hunk 1 above)
Comment (tyler-akamai) [reply packages/manager/src/components/ActionsPanel/ActionsPanel.tsx:13]: Good work!

Comment (abailly-akamai) [reply packages/manager/src/components/ActionsPanel/ActionsPanel.tsx:13]: Very nice 👍

File: packages/manager/src/features/Profile/SSHKeys/DeleteSSHKeyDialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    act(() => {
+      userEvent.click(cancelButton);
+    });
+
+    await waitFor(() => {
+      expect(props.onClose).toBeCalled();
+    });
+  });
+});
```
Comment (abailly-akamai) [inline packages/manager/src/features/Profile/SSHKeys/DeleteSSHKeyDialog.test.tsx:68]: 🎉

**Review by abailly-akamai — APPROVED** [review]
Great fixes, confirmed button labels are back. ✅ 

And agree a label should never be optional. Even if actually optional in a component, for accessibility purposes it should be required with an additional prop to visually hide it if need be.

**Pull request merged**